### PR TITLE
Password Policy & Lockout

### PR DESCRIPTION
**Pull request by kave**
@rosskarchner @kurtw 

### User flows effected
- Login
- Account password change flow
- forgot my password flow

### Test
- Type in multiple wrong passwords to be locked out
- Go through "Forgot my password" flow
- goto Account Page, and try to change your password
 - should see password policy is enforced

### CODE REVIEW

**Comment by rosskarchner** [comment]
I'll test this out later-- but it *looks* good.

File: cfgov/cfgov/urls.py [hunk 1]
```diff
@@ -6,17 +6,32 @@
 from sheerlike.views.generic import SheerTemplateView
 from sheerlike.feeds import SheerlikeFeed
 
-from v1.views import LeadershipCalendarPDFView, EventICSView, unshare, renderDirectoryPDF
+from v1.views import LeadershipCalendarPDFView, EventICSView, unshare, renderDirectoryPDF, \
```
Comment (sebworks) [inline cfgov/cfgov/urls.py]: Is `EventICSView` being used?

File: cfgov/cfgov/util.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def admin_emails(delimited_list):
+    emails = []
+
+    if delimited_list:
```
Comment (sebworks) [inline cfgov/cfgov/util.py]: Is the check necessary if you use `def admin_emails(delimited_list='')`?

File: cfgov/cfgov/settings/production.py [hunk 3]
```diff
@@ -1 +1,27 @@
 from .base import *
+
+# Sends an email to developers in the ADMIN_EMAILS list if Debug=False for errors
+LOGGING = {
+    'version': 1,
+    'disable_existing_loggers': False,
+    'filters': {
+        'require_debug_false': {
+            '()': 'django.utils.log.RequireDebugFalse'
```
Comment (sebworks) [inline cfgov/cfgov/settings/production.py]: what does the `()` key indicate?

File: cfgov/templates/wagtailadmin/login.html [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                </li>
+            </ul>
+        </form>
+    </div>
+{% endblock %}
+
+{% block extra_js %}
+{{ block.super }}
+<script>
+    $(function(){
```
Comment (sebworks) [inline cfgov/templates/wagtailadmin/login.html]: Probably would be better if we didn't use jquery here.

Comment (kave) [inline cfgov/templates/wagtailadmin/login.html]: @sebworks so this is a wagtail template which i'm overriding, i mainly edited the `fields` section everything else i left as is

File: cfgov/cfgov/urls.py (hunk 1 above)
Comment (kave) [inline cfgov/cfgov/urls.py]: hmm doesn't look like it. i'll remove it

File: cfgov/cfgov/util.py (hunk 2 above)
Comment (kave) [inline cfgov/cfgov/util.py]: yea the if only allows execution if its anything but `None` or `''`

File: cfgov/templates/wagtailadmin/account/password_reset/confirm.html [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+{% block extra_css %}
+    {{ block.super }}
+
+    {% compress css %}
+        <link rel="stylesheet" href="{% static 'wagtailadmin/css/layouts/login.css' %}" type="text/css" />
+    {% endcompress %}
+{% endblock %}
+
+{% block furniture %}
```
Comment (kurtw) [inline cfgov/templates/wagtailadmin/account/password_reset/confirm.html]: > block furniture
lol wut

File: cfgov/cfgov/settings/production.py (hunk 3 above)
Comment (kave) [inline cfgov/cfgov/settings/production.py]: not exactly sure, but its part of the syntax they use to instantiated filters https://docs.djangoproject.com/en/1.9/topics/logging/#django.utils.log.RequireDebugFalse

File: cfgov/templates/wagtailadmin/account/password_reset/confirm.html (hunk 5 above)
Comment (kave) [inline cfgov/templates/wagtailadmin/account/password_reset/confirm.html]: yea some of these blocks names are questionable lol

**Comment by rosskarchner** [comment]
This is pretty bad-ass, @kave.

One weird bit, is that the "passwords don't match" error appears differently than the password quality errors. Can we get that one to appear in the same error box?

**Comment by sebworks** [comment]
@kave, thanks for he explanations.

**Comment by kurtw** [comment]
I'll +1 what Ross said. :facepunch: :anger: :tada:

**Comment by rosskarchner** [comment]
Just noticed one thing-- a user can, once logged in, navigate to /django-admin, which has it's own password change screen that doesn't obey the new rules. Maybe we need to capture that URL, and redirect to our new form? Does wagtail have a similar page for already logged-in users to change their password?


screen shot 2016-01-20 at 9 38 12 am

**Comment by kave** [comment]
@rosskarchner ahh i see. yea it does at the `/account/change_password/` url. i can reroute the django admin url to that

**Comment by rosskarchner** [comment]
Similarly, we probably need to intercept people who end up at http://localhost:8000/django-admin/login/ , maybe involves setting these?

https://docs.djangoproject.com/en/1.9/ref/settings/#login-redirect-url

**Comment by kave** [comment]
@rosskarchner so creating those setting fields didn't seem to work as expected, so i just overridden the admin url paths https://github.com/cfpb/cfgov-refresh/commit/c33030ceb5f5422dbe623d534c831fe15030caaf#diff-e45111e649c8f0371d5367b7351f1dd8R21

**Comment by rosskarchner** [comment]
:+1:

**Pull request merged**